File: c++/RecursiveStringReverse.cpp
#include<iostream>
using namespace std;

void rev(string str){
    if(str.length() == 0)
        return;

    rev(str.substr(1));
    cout<<str[0];
}

int main(){
    string str = "Algorithms";
    rev(str);
    return 0;
}

File: java/RecursiveStringReverse.java
public class Algorithms {
  void rev(String str){
    if ((str==null)||(str.length() <= 1)) 
        System.out.println(str); 
    else { 
        System.out.print(str.charAt(str.length()-1)); 
        rev(str.substring(0,str.length()-1)); 
    } 
  }
  public static void main(String[] args) {
     rev("algorithms");
  }
}
  
